src/LinearAlgebra/SparseMatrix/COO_Matrix.hpp

```diff
@@ -623,6 +627,75 @@ inline void COO_Matrix<ScalarT, IdxT>::printMatrix(std::string name)
   std::cout << std::flush;
 }
 
+/**
+ * @brief Print matrix to file in Matrix Market format
+ *
+ * @tparam ScalarT
+ * @tparam IdxT
+ *
+ * @param[in] filename Output file path
+ * @param[in] comment Optional comment for the Matrix Market file
+ */
+template <class ScalarT, typename IdxT>
+inline void COO_Matrix<ScalarT, IdxT>::printMatrixMarket(const std::string& filename, const std::string& comment)
+{
+  if (this->sorted_ == false)
+  {
+    this->sortSparse();
+  }
+
+  std::ofstream outfile(filename);
+  if (!outfile.is_open())
+  {
+    std::cerr << "Error: Unable to open file " << filename << " for writing." << std::endl;
+    return;
+  }
+
+  // Write Matrix Market header
+  outfile << "%%MatrixMarket matrix coordinate ";
+
+  // Determine real/integer/pattern type
+  if (std::is_same<ScalarT, int>::value || std::is_same<ScalarT, long>::value)
+    outfile << "integer ";
+  else if (std::is_floating_point<ScalarT>::value)
+    outfile << "real ";
+  else
+    outfile << "pattern ";
```

Comment:
What is the purpose of this? We know the matrix is of real type. Perhaps just have a check 
```c++
if (!std::is_floating_point<ScalarT>::value)
{
  // error message
  return;
}
```